internal/service/alertconfiguration/resource_alert_configuration_migration_test.go

```diff
@@ -142,3 +143,48 @@ func TestMigConfigRSAlertConfiguration_withEmptyOptionalAttributes(t *testing.T)
 		},
 	})
 }
+
+func TestMigConfigRSAlertConfiguration_withDataDog(t *testing.T) {
+	var (
+		projectID = acc.ProjectIDExecution(t)
+		ddAPIKey  = dummy32CharKey
+		ddRegion  = "US"
+
+		groupNotificationMap = map[string]string{
+			"type_name":    "GROUP",
+			"interval_min": "5",
+			"delay_min":    "0",
+		}
+
+		datadogNotificationMap = map[string]string{
+			"type_name":       "DATADOG",
+			"interval_min":    "5",
+			"delay_min":       "0",
+			"datadog_api_key": ddAPIKey,
+			"datadog_region":  ddRegion,
+		}
+
+		config = configWithDataDog(projectID, ddAPIKey, ddRegion, true, datadogNotificationMap, groupNotificationMap)
+	)
+
+	// 1.20.0 introduced handling of a breaking change from the API which required notification.#.integration_id to be
+	// updated to Option/Computed from an Optional attribute. This impacted only notifications with integrations.
+	mig.SkipIfVersionBelow(t, "1.20.0")
+
+	resource.ParallelTest(t, resource.TestCase{
```

Comment:
mig and acc test can be reused with a helper function, e.g.: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/searchindex/resource_search_index_test.go#L16

Reply:
updated